scripts/check_compliance.py

```diff
@@ -29,6 +29,10 @@ def info(what):
 def error(what):
     sys.stderr.write(Fore.RED + what + Style.RESET_ALL + "\n")
 
+def warning(what):
```

Comment:
this is the actual "user interface", has nothing to do with logging. basically printing messages in different colours on the screen.

Reply:
OK, then I guess all I'm asking is moving that answer to a comment on top of the block of these info()/warn()/error() functions so people understand the difference with logging.info()/.warn()/.error()

Reply:
I got rid of `error()` and `info()` in https://github.com/zephyrproject-rtos/ci-tools/pull/23 before I saw this. Could add them back though.

I think it's a bit confusing to have e.g. `error()`, `logging.error()`, and `ComplianceTest.error()` too.

Personally, I'd just get rid of all the `logging` stuff and use manual `print()` debugging, since it's a pretty simple script, and it's nice to have it readable. Would be nice to get rid of one of them at least.

Reply:
Removed the commit that removed `error()` and `info()` from https://github.com/zephyrproject-rtos/ci-tools/pull/23, so that it doesn't get stuck on that.

Reply:
yeah, I am not sure what the logging is good for either tbh. Will rename those function and make them clearer.

Reply:
logging is good for debugging the script with an adjustable log level (without reinventing the built-in module)

Reply:
To clarify "debugging" is not just for the 3 persons changing this script, I should have found a different word. It's for any user wondering what's going on, think -v -v -v.

Reply:
Here's one possible/typical interaction between logging and a --dry-run option (_any_ dry run option):
=> always log .info()/.debug() what is going on, not just when --dry-run. Because why not?
```python
get_comments_from_github(); 
maybe_some_logging()
parse_comments()
maybe_some_logging()
prepare_comment_changes()
maybe_some_logging()
# only instance of this test
# = one or two lines total difference between dry run and real
if not dry_run: 
    actually_write_and_commit_comment_changes_for_real()
```
When doing it like this, a dry-run option is 99% identical to the real thing. The current --dry-run submission is far from that.

If desired:
```python
if dry_run and loglevel > info:
    logging.warn("that's not very useful")
````

> manual print() debugging, since it's a pretty simple script, and it's nice to have it readable.

Can you elaborate why print() is more readable than logging.debug() ?